web/src/components/ToggleTheme.astro

```diff
@@ -0,0 +1,111 @@
+---
+import SunIcon from "./icons/sun.astro"
+import MoonIcon from "./icons/moon.astro"
+import SystemIcon from "./icons/system.astro"
+
+const THEMES = ["Light", "Dark", "System"]
+---
+
+<div class="relative">
+  <button
+    id="theme-toggle-btn"
+    class="appearance-none border-none flex hover:scale-110 transition"
+  >
+    <span class="sr-only">Choose a theme</span>
+    <SunIcon id="light" class="theme-toggle-icon size-4 transition-all" />
+    <MoonIcon
+      id="dark"
+      class="theme-toggle-icon absolute size-4 transition-all" />
+    <SystemIcon
+      id="system"
+      class="theme-toggle-icon absolute size-4 transition-all" />
+  </button>
+  <div
+    id="themes-menu"
+    class="absolute opacity-0 scale-80 top-8 right-0 text-sm p-1 min-w-[8rem] rounded-md border border-gray-100 bg-white/90 dark:bg-zinc-900/70 dark:border-gray-500/20 shadow-[0_3px_10px_rgb(0,0,0,0.2)] backdrop-blur-md z-50"
```

Comment:
@ivanosquis10 I would recommend a `bg-zinc-900` as almost full black might not be best for dark mode.

Reply:
> @ivanosquis10 I would recommend a `bg-zinc-900` as almost full black might not be best for dark mode.

you say in the layout?

Reply:
This one is fine..